docs/team/limerencee.adoc

```diff
@@ -35,21 +40,140 @@ _/src/java/thrift/logic/Logic.java_:: Italic text indicates that it is a file pa
+* `**Value**` is parsed by `AddTransactionCommandParser#parseTransactionValue(ArgumentMultimap)`
+* `**TransactionDate**` is instantiated by `AddTransactionCommandParser#parseTransactionDate()`
+
+NOTE: `**ArgumentMultimap**` is a class that stores all the parsed parameters taken from the user input.
+
+Since the user input is valid, the `**Expense**` is successfully created and inserted into the transaction list.
+The transaction list now contains 1 `**Transaction**` object.
+
+image::add-transaction/AddTransaction1.png[,350]
+
+Step 2. The user executes **add_income n/Bursary v/500 r/Awarded for doing well in school** to insert an `**Income**`.
+The input is now checked in a similar fashion as in Step 2 except for:
+
+* `**Remark**` is parsed by `AddTransactionCommandParser#parseTransactionRemark(ArgumentMultimap)`
+
+Again, since the input is valid, the `**Income**` is successfully added into the transaction list. The transaction list
+now contains 2 `**Transaction**` objects.
+
+image::add-transaction/AddTransaction2.png[,350]
+
+The following activity diagram summarizes what happens when the user executes a command to add a new `**Transaction**`:
+
+.Activity diagram of adding a `**Transaction**` into the transaction list
+image::add-transaction/AddTransactionActivity.png[, 500]
+
+==== Design considerations
+There are many different ways to implement how a transaction is added into `**THRIFT**`. In this section, we will be
+justifying why we chose to implement it the way we did.
+
+===== Aspect: Differentiating between `**Expense**` and `**Income**`
+* **Alternative 1: (current choice):** Introduce a `**Transaction**` parent class which both `**Expense**` and `**Income**`
+extends from.
+** Pros: Introduces polymorphism, easing references to either classes by simply referencing the `**Transaction**` object.
+For example, using a single `**List<Transaction>**` instead of needing 2 separate lists `**List<Expense>**` and `**List<Income>**`.
+** Cons: Reduces the readability of the program as polymorphism can be confusing.
+
+* **Alternative 2:** Keep `**Expense**` and `**Income**` classes separate, with each having their own specialized methods.
+** Pros: Maintains an intuitive design: `**Expense**` deducts money and `**Income**` increases money.
+** Cons: Incurs significant overhead and duplicated codes since it is likely that both `**Expense**` and `**Income**` will
+have very similar methods.
+
+Alternative 1 was chosen because we want to model it close to the real world: both `**Expense**` and `**Income**` are described
+as being a `**Transaction**`.
+
+===== Aspect: Managing how `**Value**` is stored and handled in `**Expense**` and `**Income**`
+* **Alternative 1: (current choice):** Disallow negative `**Value**` in `**Expense**` object, only using positive amount
+for both `**Expense**` and `**Income**`
+** Pros: Removes the need to implement support for inserting negative `**Value**`. This is due to how `**Value**` constraints
```

Comment:
Violates the parallelism principle, you use verbs here on alternative 1 but used nouns on alternative 2.